code/parse/parselo.cpp

```diff
@@ -2327,11 +2327,156 @@ void coerce_to_utf8(SCP_string &buffer, const char *str)
 	Warning(LOCATION, "Truncating non-UTF-8 string '%s' to '%s'!\n", str, buffer.c_str());
 }
 
+//Converts a string in ISO 8859-1 to the currently used encoding
+//Only legal for strings with correct ISO 8859-1 encoding
+SCP_string iso8859_1_to_current(const char* str)
+{
+	SCP_string buffer;
+	//We are not in unicode mode, so just return as is, the string is correct
+	if (!Unicode_text_mode) {
+		buffer.assign(str);
+		return buffer;
+	}
+
+	//If not, convert to utf-8
+	auto len = strlen(str);
+
+	// Validate the UTF-8 encoding
+	bool valid = true;
+
+	for (size_t i = 0; i < len; i++) {
+		if (str[i] > 0x79 || str[i] < 0) {
+			valid = false;
+			break;
+		}
+	}
```

Comment:
This would be useful to put in its own function.

However, isn't it possible to get the same result with `utf8::find_invalid`?

Reply:
utf8::find_invalid is not the same as this does here. What find_invalid does, it figure out if the sequence is a correct utf8 string. What this does is check, if the sequence only contains chars between 0 and 0x79. This is because in that special case, a utf8 string and an iso8859-1 string are equal and no change needs to be done. If other chars exist, they can be valid utf8, but not valid iso8859-1. Thus this different check is necessary. But I will change it to it's own function.
As a sidenote: A valid iso8859-1 string that does not fulfill the above check is not valid utf8 (I think), so in that direction the utf-8 check would work, but this more stringent check is definitely needed for the utf8 to iso8859-1 conversion